cli/pkg/local/app.go

```diff
@@ -264,14 +264,6 @@ func (a *App) Serve(httpPort, grpcPort int, enableUI, openBrowser, readonly bool
 		Readonly:         readonly,
 	}
 
-	// Create server logger.
-	// It only logs error messages when !verbose to prevent lots of req/res info messages.
-	lvl := zap.ErrorLevel
-	if a.Verbose {
-		lvl = zap.DebugLevel
-	}
-	serverLogger := a.BaseLogger.WithOptions(zap.IncreaseLevel(lvl))
```

Comment:
Instead of making log levels on the server configurable (by passing it into `LoggingUnaryServerInterceptor`, `ServeGRPC`, etc.), I think we should keep using info logs in the server and then keep this snippet here instead.

Reply:
I think I removed this because the logic is duplicated in `initLogger` function [here](https://github.com/rilldata/rill/blob/main/cli/pkg/local/app.go#L480). Also If we keep info level for grpc calls then it will start to show in console logs which were not being shown earlier, hence passing debug level for local rill-dev runtime and info level for standalone/cloud runtime.

Reply:
This code is slightly different – basically what it does is it increases the level of the server logger to error, which means anything below error will be discarded. Therefore, the server can print at info level without anything being output on local:
```go
// a.BaseLogger outputs at info-level
serverLogger := a.BaseLogger.WithOptions(zap.IncreaseLevel(zap.ErrorLevel))
// a.BaseLogger still outputs at info-level
// But serverLogger only forwards error+ level logs to a.BaseLogger
```

Reply:
Went through the code again. We have parent logger created in `NewApp`, that logger has two cores - file and console and depending on verbose flag the level will be either debug or info. Now this parent logger is given to runtime and controller creates a new duplicating logger from this. This logger is also given to server which by defaults logs only error however with verbose it can log at debug level because parent logger (a.BaseLogger) will also have debug level set. 

So few observations - 
1. Irrespective of verbose flag, server grpc logs goes only to file never console as they are not console namespaced.
2. When verbose is set the code above `zap.IncreaseLevel(zap.DebugLevel)` is essentially a no-op as BaseLogger already have debug level so we can remove this and simplify.

Now in the current PR -
Few observations -
1. If we keep the code above then server logger will allow debug+ level logs and the since the console core of the parent logger allows info+, all the info level grpc calls will go to console which is different behaviour from earlier but seems ok. This was the reason to send explicit flag. Just an observation we can keep the code.
2. While creating the parent logger, console core can be created directly using `zapcore.NewCore(consoleEncoder, zapcore.Lock(os.Stdout), zapcore.InfoLevel)` instead of `zapfilter.NewFilteringCore(zapcore.NewCore(consoleEncoder, zapcore.Lock(os.Stdout), logLevel), zapfilter.MinimumLevel(zapcore.InfoLevel))` since even if debug level is set for parent, console will not get them because of `zapfilter.MinimumLevel(zapcore.InfoLevel)` rule.

Reply:
> 2. When verbose is set the code above zap.IncreaseLevel(zap.DebugLevel) is essentially a no-op as BaseLogger already have debug level so we can remove this and simplify.

It's not quite a no-op because it changes the server logger level from error (as set on line 269) to debug. But you're right it could have been simplified slightly like this:
```go
serverLogger := a.BaseLogger
if !a.Verbose {
  serverLogger = a.BaseLogger.WithOptions(zap.IncreaseLevel(zap.ErrorLevel))
}
```

> 1. If we keep the code above then server logger will allow debug+ level logs and the since the console core of the parent logger allows info+, all the info level grpc calls will go to console which is different behaviour from earlier but seems ok. This was the reason to send explicit flag. Just an observation we can keep the code.

By default, `a.Verbose` is false, so all the server-level info logs will go neither to the console nor the log file. That's what we want since server-level info logs are very chatty and not really useful on local.

But if `a.Verbose` is true, then after this change, it will be as you say that the server info logs will go to the console. And I think that's an improvement – because if you pass `--verbose` it's better to output as much as possible to the console.

> 2. While creating the parent logger, console core can be created directly using zapcore.NewCore(consoleEncoder, zapcore.Lock(os.Stdout), zapcore.InfoLevel) instead of zapfilter.NewFilteringCore(zapcore.NewCore(consoleEncoder, zapcore.Lock(os.Stdout), logLevel), zapfilter.MinimumLevel(zapcore.InfoLevel)) since even if debug level is set for parent, console will not get them because of zapfilter.MinimumLevel(zapcore.InfoLevel) rule.

Yeah looks like this can be simplified. But one difference here – if `a.Verbose` is true, we should change the console log level to `zapcore.DebugLevel`. Since as stated above, if someone passes `--verbose` we just want to output as much as possible to the console.